File: algorithms/1_easy/P/00507_perfect_number/solution.py
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        if num <= 1: return False

        total_sum = 1
        sq = int(sqrt(num))

        for i in range(2, sq + 1):
            if num % i == 0:
                total_sum += i

                other = num // i
                if i != other:
                    total_sum += other

                if total_sum > num:
                    return False

        return total_sum == num

File: algorithms/1_easy/P/00507_perfect_number/solution.c
bool checkPerfectNumber(const int num) {
    if (num <= 1) { return false; }

    int sum = 1;
    const int sq = (int)sqrt(num);

    for (int i = 2; i <= sq; i += 1) {
        if (num % i == 0) {
            sum += i;

            const int other = num / i;
            if (i != other) {
                sum += other;
            }

            if (sum > num) {
                return false;
            }
        }
    }

    return sum == num;
}
